.github/workflows/static.yml

```diff
@@ -0,0 +1,94 @@
+name: golang/static
+on: pull_request
+
+jobs:
+  fmt: 
+    name: fmt
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: fmt
+      uses: grandcolline/golang-github-actions@v1.1.0
```

Comment:
Can we pin the version using full commit SHA:
```suggestion
      uses: grandcolline/golang-github-actions@d1fab268ef4df10e7e11f248a179fedf9e94e054
```

Reply:
Version seems more human friendly and reliable, why use a sha?

Reply:
Anybody with write access to the repo housing the action (in this case `grandcolline/golang-github-actions`) can make the git version tag point at arbitrary code. We pass `secrets.GITHUB_TOKEN` to the action, so it has lots of powers:
> That includes read/write access to code, commit statuses, repository projects, collaborators, and deployment statuses for public repositories and organizations.

Reply:
Understood, but this is a cure worse than the poison, why don't we fork instead?

Reply:
https://www.zdnet.com/article/sha-1-collision-attacks-are-now-actually-practical-and-a-looming-danger/

Reply:
> this is a cure worse than the poison  

That seems a bit extreme 😉. We can just add a comment saying "d1fab268ef4df10e7e11f248a179fedf9e94e054 is v1.1.0"?

> why don't we fork instead?

That also works, but introduces some follow-on headaches around access control to the forked repo.

Reply:
> We can just add a comment saying "d1fab268ef4df10e7e11f248a179fedf9e94e054 is v1.1.0"

Sure this is better, but my understanding is that it's pretty easy to construct a replacement SHA if they want to attack us.

Reply:
It is not easy. 

1) There is no known second preimage attack against SHA1, which is what you'd need to retroactively find a commit with the same commit hash.

2) This leaves premeditated collision attacks. Git (& GitHub) use https://eprint.iacr.org/2017/173 (implementation [here](https://github.com/cr-marcstevens/sha1collisiondetection)). AFAIK, there are no known collision attacks against SHA1 which break the Stevens & Shumow scheme.

Of course, that doesn't invalidate your point that SHA1 is broken in principle. But I would be comfortable with simply pinning hashes for the time being and dealing with proper vendoring a bit later, when we have collected more experience around GitHub Actions. 🤷‍♂️

Reply:
Should we also pin to a commit on the `actions/checkout` action?  Sure, it's published by Github, but that's still a vulnerability.

Reply:
We use these all over the place, probably should be an audit.

Reply:
It's just the formatting check, is there really an attack vector here?

Not trolling, genuinely curious. Wouldn't the worst an attacker can do be to allow us to merge some unformatted code?

Reply:
Ah, just re-read your comment @kaleofduty .

In that case it seems like the appropriate action would be to restrict the permissions of `secrets.GITHUB_TOKEN`.

Reply:
The default list of permissions is indeed very permissive.

https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

Reply:
@kaleofduty another option is to keep the code for actions in our own repo (simple copy or submodule).

Do you think this would be better?

Reply:
I prefer this personally.

Reply:
Downsides of forking the action's repo:
- We have even more repos under our org
- We need to set permissions for the action's repo to be at least as strict as those of the repo in which the action is used. We need to do this per forked repo. How do we deal with the case of the same action being used in different repos that have different permissions?
- We need to keep the fork in sync with releases from upstream.

Reply:
Will set comment: false then it doesnt requiire a token at all.